src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Functional.Tests
+{
+    /// <summary>
+    /// Provides a test-only, overly-simplified HTTP proxy for a single GET request.  It is not meant to be robust,
+    /// but simply for verification that a GET request is in fact getting routed through a designated proxy
+    /// and providing credentials if expected.
+    /// </summary>
+    internal sealed class LoopbackGetRequestHttpProxy
+    {
+        public struct ProxyResult
+        {
+            public byte[] ResponseContent;
+            public string AuthenticationHeaderValue;
+        }
+
+        public static Task<ProxyResult> StartAsync(out int port, bool requireAuth)
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+            listener.Start();
+            port = ((IPEndPoint)listener.LocalEndpoint).Port;
+            return StartAsync(listener, requireAuth);
+        }
+
+        private static async Task<ProxyResult> StartAsync(TcpListener listener, bool requireAuth)
+        {
+            ProxyResult result = new ProxyResult();
+            var headers = new Dictionary<string, string>();
+            Socket clientSocket = null;
+            Stream clientStream = null;
+            StreamReader clientReader = null;
+            string url = null;
+            try
+            {
+                // Get and parse the incoming request
```

Comment:
nit: add period at end of sentence.  There are some other comments in this file that need periods also.